packages/jest-environment-node/src/index.ts

```diff
@@ -58,6 +58,7 @@ class NodeEnvironment implements JestEnvironment {
     ) {
       global.TextEncoder = TextEncoder;
       global.TextDecoder = TextDecoder;
+      global.Uint8Array = Uint8Array;
```

Comment:
This needs its own if block to check whether `typeof Uint8Array !== 'undefined'` is true. Feel free to also leave a comment on which Node version supports it, like it's done for other globals

Reply:
@thymikee that's not a bad idea!

My initial thinking was that this is only a workaround specific to the `TextDecoder ` as it seems to be referencing a different `Uint8Array` constructor, so I grouped them all together in the same block. If `TextEncoder` isn't defined then re-assigning `Uint8Array` isn't really needed.

```js
// in Jest (using the node environment)
new TextEncoder().encode('').constructor === Uint8Array // false

// outside Jest (regular node runtime)
new TextEncoder().encode('').constructor === Uint8Array // true
```

Could you please help understand under what conditions `Uint8Array` would be `undefined`? I think this helps me better understand the Jest architecture =)

`Uint8Array` is very well supported since node **`v0.10`**, unlike `TextEncoder` that was only made available in the global namespace with node **`v11.0.0`** (which is relatively recent). I'm not sure if there will be a case where `Uint8Array` isn't defined (`v0.10` dates back to 2013 or maybe even before that).

Reply:
We should be good with adding `Uint8Array` next to the `ArrayBuffer` then, wdyt @SimenB?

Reply:
> If TextEncoder isn't defined then re-assigning Uint8Array isn't really needed.

Hey hey! this conversation made me realize that this issue does not only occur with the global `TextEncoder` (in Node v11), **but also with `require('util').TextEncoder` (in any version of node)**! 😬 

In Node v10 (or older), this test would still fail for the same reason:

```js
// node v10 (importing TextEncoder from the 'util' module)
const {TextEncoder} = require('util');

test('TextEncoder.encode references the same global Uint8Array constructor', () => {
  expect(new TextEncoder().encode()).toBeInstanceOf(Uint8Array)
});
```

```
  ✕ TextEncoder.encode references the same global Uint8Array constructor (5ms)

  ● TextEncoder.encode references the same global Uint8Array constructor

    expect(received).toBeInstanceOf(expected)

    Expected constructor: Uint8Array
    Received constructor: Uint8Array
```

Looks like after all we actually need to assign `global.Uint8Array` regardless of `TextEncoder` being global or not to support both cases 😅..

I'll go ahead and push the changes 👍